src/sentry/models/release.py

```diff
@@ -816,3 +816,24 @@ def safe_delete(self):
             releasefile.file.delete()
             releasefile.delete()
         self.delete()
+
+    @classmethod
+    def with_artifact_counts(cls, *args, **kwargs):
+        # Have to exclude Releases without release files because COALESCE would
+        # give them an artifact count of 1
```

Comment:
Should the coalesce fallback to 0 instead then?

Reply:
I want the `artifact_count` value `None` on a release file to mean 1 semantically. But when there are no release files, the sum should be 0.

The simpler solution would be to set `artifact_count` to 1 for all existing release files, but I assumed this would make the migration too expensive.

Reply:
Yeah, unfortunately at 5.7B rows we can't realistically backfill this table. I think assuming null = 1 is ok here.